### PR TITLE
Feature/export refactor 2

### PR DESCRIPTION
**Pull request by czpython**
* adds export wizard
* moves old export logic into the old form data admin.
* packages some common functionality in forms admin base
* changes behavior to not store anything as a label if no label was defined.
* adds new export logic. The ``exporter.py`` should be cleaned up a bit.

### CODE REVIEW

File: aldryn_forms/admin/formdata.py [hunk 1]
```diff
@@ -20,5 +30,89 @@ class FormDataAdmin(BaseFormSubmissionAdmin):
...
+
+                try:
+                    # Since django-tablib 3.1 the parameter is called file_type
+                    response = do_export(file_type='xls')
+                except TypeError:
+                    response = do_export(format='xls')
+                return response
+            else:
+                self.message_user(request, _("No records found"), level=messages.WARNING)
+                export_url = 'admin:{}'.format(self.get_admin_url('export'))
```
Comment (cyberstar) [inline aldryn_forms/admin/formdata.py]: if it should work on python 2.6 please use indexes for string formatting like ``{0}``.

File: aldryn_forms/admin/forms.py [hunk 2]
```diff
@@ -54,14 +63,18 @@ def clean(self):
...
-        filename = self.export_filename.format(
+        filename_format = self.export_filename.format(
             form_name=slugify(form_name),
             language=data['language'],
         )
-        return timezone.now().strftime(filename)
+        filename = timezone.now().strftime(filename_format)
+
+        if extension:
+            filename = '{}.{}'.format(filename, extension)
```
Comment (cyberstar) [inline aldryn_forms/admin/forms.py]: same thing about python 2.6 and string formatting

File: aldryn_forms/admin/views.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            kwargs['submissions'] = form.get_queryset()
+        return kwargs
+
+    def render_next_step(self, form, **kwargs):
+        next_step = self.steps.next
+
+        if next_step == self.steps.last and not form.get_queryset().exists():
+            self.storage.reset()
+            self.admin.message_user(self.request, ugettext("No records found"), level=messages.WARNING)
+            export_url = 'admin:{}'.format(self.admin.get_admin_url('export'))
```
Comment (cyberstar) [inline aldryn_forms/admin/views.py]: same for string formatting

File: aldryn_forms/models.py [hunk 4]
```diff
@@ -30,18 +31,51 @@
...
 
 
+class SerializedFormField(BaseSerializedFormField):
+
+    @property
+    def field_id(self):
+        field_label = self.label.strip()
+
+        if field_label:
+            field_as_string = u'{}-{}'.format(field_label, self.field_type)
```
Comment (cyberstar) [inline aldryn_forms/models.py]: same for string formatting

File: aldryn_forms/models.py [hunk 5]
```diff
@@ -30,18 +31,51 @@
...
+        else:
+            field_as_string = self.name
+        field_id = u'{}:{}'.format(field_as_string, self.field_occurrence)
```
Comment (cyberstar) [inline aldryn_forms/models.py]: same for string formatting

File: aldryn_forms/models.py [hunk 6]
```diff
@@ -477,17 +532,33 @@ class Meta:
     def __unicode__(self):
         return self.name
 
-    def _form_data_hook(self, data):
+    def _form_data_hook(self, data, occurrences):
+        field_label = data['label'].strip()
+
+        if field_label:
+            field_type = data['name'].rpartition('_')[0]
+            field_id = u'{}_{}'.format(field_type, field_label)
```
Comment (cyberstar) [inline aldryn_forms/models.py]: same for string formatting

**Comment by cyberstar** [comment]
LGMT except for string formatting, also you can get rid of all the u'' with ``from __future__ import unicode_literals``

**Comment by czpython** [comment]
Python 2.6 support was dropped a few months ago.

**Pull request merged**